src/chemistry/aerosol/modal_aerosol_state_mod.F90

```diff
@@ -14,7 +14,6 @@ module modal_aerosol_state_mod
 
   type, extends(aerosol_state) :: modal_aerosol_state
      private
-     type(physics_state), pointer :: state => null()
```

Comment:
Instead of removing this physics_state pointer data member, why not change this to a pointer to a real array, such as:
```
  real(r8), pointer :: state_array(:,:,:)
```
Then we would not need to change the interfaces of the member procedures, as you did.

In my mind, an aerosol_state object would encapsulate the aerosol state data and one would not need to pass (parts of) the state of the aerosols to its member procedures.

Curiously, you left the physics buffer pointer member function, so this is confusing.

Maybe it  is best to explain in person.

Reply:
The state will not be a pointer in the future.

Reply:
The reason for pulling the constituent array out to the calling interface is that this array belongs to the host model and is not private to the aerosol object. Also, it will not be a pointer in the future so your proposal will not work.
I left the pbuf alone because that data will become part of the aerosol object in the future so it should not be in the calling interface.

Does this make sense?